bundles/org.openhab.binding.modbus.stiebeleltron/src/main/java/org/openhab/binding/modbus/stiebeleltron/internal/handler/StiebelEltronHandler.java

```diff
@@ -0,0 +1,736 @@
+
+    /**
+     * Reference to the modbus manager
+     */
+    protected ModbusManager modbusManager;
+
+    /**
+     * Instances of this handler should get a reference to the modbus manager
+     *
+     * @param thing the thing to handle
+     * @param modbusManager the modbus manager
+     */
+    public StiebelEltronHandler(Thing thing, ModbusManager modbusManager) {
+        super(thing);
+        this.modbusManager = modbusManager;
+    }
+
+    /**
+     * @param address address of the value to be written on the modbus
+     * @param shortValue value to be written on the modbus
+     */
+    protected void writeInt16(int address, short shortValue) {
+        @Nullable
+        StiebelEltronConfiguration myconfig = StiebelEltronHandler.this.config;
+        if (myconfig == null) {
+            throw new IllegalStateException("registerPollTask called without proper configuration");
+        }        
+        // big endian byte ordering
+        byte b1 = (byte) (shortValue >> 8);
+        byte b2 = (byte) shortValue;
+
+        ModbusRegister register = new BasicModbusRegister(b1, b2);
+        ModbusRegisterArray data = new BasicModbusRegisterArray(new ModbusRegister[] { register });
+
+        BasicModbusWriteRegisterRequestBlueprint request = new BasicModbusWriteRegisterRequestBlueprint(slaveId,
+                address, data, false, myconfig.getMaxTries());
+
+        ModbusSlaveEndpoint slaveEndpoint = this.endpoint;
+        if (slaveEndpoint == null) {
+            return;
+        }
+
+        BasicWriteTask writeTask = new BasicWriteTask(slaveEndpoint, request, new ModbusWriteCallback() {
+            @Override
+            public void onWriteResponse(ModbusWriteRequestBlueprint request, ModbusResponse response) {
+                if (hasConfigurationError()) {
+                    return;
+                }
+                logger.debug("Successful write, matching request {}", request);
+                StiebelEltronHandler.this.updateStatus(ThingStatus.ONLINE);
+            }
+
+            @Override
+            public void onError(ModbusWriteRequestBlueprint request, Exception error) {
+                StiebelEltronHandler.this.handleError(error);
+            }
+        });
+        logger.trace("Submitting write task: {}", writeTask);
+        modbusManager.submitOneTimeWrite(writeTask);
+    }
+
+    /**
+     * @param command get the value of this command.
+     * @return short the value of the command multiplied by 10 (see datatype 2 in the stiebel eltron modbus
+     *         documentation)
+     */
+    private short getScaledInt16Value(Command command) {
+        if (command instanceof QuantityType) {
+            QuantityType<?> c = (QuantityType<?>) command;
+            return (short) (c.doubleValue() * 10);
+        }
+        if (command instanceof DecimalType) {
+            DecimalType c = (DecimalType) command;
+            return (short) (c.doubleValue() * 10);
+        }
+        return 0;
```

Comment:
add the "throw IllegalArgumentException"

Reply:
It's bad practice to throw unchecked exceptions (as `IllegalArgumentException` is) on expected errors, since the compiler doesn't enforce you to catch those. Better create your own exception, which extends from `Exception` and throw that.

Reply:
Changed exception type to StiebelEltronException